Documentation/proposals/202309-controller-id.md

```diff
@@ -0,0 +1,79 @@
+# Controller-ID Proposal
+
+* Owners:
+  * [danielmellado](https://github.com/danielmellado)
+* Related Tickets:
+  * [#4281](https://github.com/prometheus-operator/prometheus-operator/issues/4281)
+  * [#4498](https://github.com/prometheus-operator/prometheus-operator/pull/4498)
+* Other Docs:
+  * n/a
+
+# Introduction
+
+This proposal aims to implement a solution to support multiple cluster-level
+Prometheus instances running concurrently without conflicting over the same
+custom resources. This solution isn't limited to the Prometheus resources as
+it'll also be available for `AlertManager`and `ThanosRuler` ones, as well as for
+any pod-based resource that could be added in the future.
+
+This issue can significantly impact use cases where multiple Prometheus operator
+instances run at the same time in the Kubernetes cluster.
+
+# Why
+
+Currently, we encounter issues when different users deploy different instances
+of the Prometheus operator, that will try to reconcile the same resources.
+
+In the worst-case scenario, these operators may not only compete for ownership
+of the CRD resources but also attempt to rewrite or redeploy different versions
+of the CRD, causing disruptions to all pods.
+
+The remediation for this scenario, where users deploy their Prometheus operator
+instances in parallel, involves using one of the many CLI arguments such as
+`--deny-namespaces`, `--namespaces`, `--prometheus-instance-selector`  or
+`prometheus-instance-namespaces`. But this requires cooperation between the
+different parties and there's no way to ensure that a specific monitoring
+resource is managed only by a specific operator instance.
+
+# How
+
+After some research, we have identified a potential solution already
```

Comment:
I forgot that we should give credit to @machine424 for the suggestion :)
```suggestion
After some research from @machine424, we have identified a potential solution already
```

Reply:
Totally agree, adding him back there, wasn't sure whether it should be on the proposal.